### ISSUE TITLE
Feature request: show fee in fiat in Send tab

### ISSUE THREAD

**Issue by charleshsu168** [spesmilo/electrum#2838]
I am using 2.8.3
please include option to show fee in USD next to the slider bar.
currently only showing fee in btc. not very helpful for those of us who think fee in terms of dollars. 
thank you.
![untitled](https://user-images.githubusercontent.com/5799190/30038724-a911eda6-917c-11e7-93cd-480a55e02173.png)

**Comment by pmukherjeee** [comment]
Yeah, this is useful. I second this.

**Comment by cdodd** [comment]
Or more usefully, display it in the fiat currency that the user has selected in the preferences.

**Comment by ecdsa** [comment]
the fiat fee is already shown in the confirmation dialog.
how many times should it be shown?

**Comment by Engelberg** [comment]
@ecdsa Ideally, it should be shown on the page with the slider, where you are selecting what level of fee to select, before you click send.  So putting the fiat fee as specified in the picture above would be superior to putting it in the confirmation dialog (which the user has no reason to assume will appear after hitting send, with a fiat fee given).

**Comment by Engelberg** [comment]
My preference would be to show the fiat fee to the right of the bitcoin fee in the above photo, rather than having it *instead* of the bitcoin fee.  Both are useful to see.

**Comment by pydlv** [comment]
@ecdsa I don't see any downside to showing the fee in fiat on this screen. Support.

**Comment by fresheneesz** [comment]
I also want to reference this feature request, which partially duplicates this one: https://github.com/spesmilo/electrum/issues/2788 . The alignment of the amount and fee fields is pretty confusing. At very least, the fee slider and the BTC fee field should swap positions.

**Comment by jonathancross** [comment]
Displaying the fee in fiat (next to the fee input) would be useful to prevent nasty surprises, but I'd suggest making it read-only.  Proper fees for a bitcoin transaction can only be calculated in BTC. If the fee seems too high, the user might choose to simply delay a transaction until there is less network congestion.

**Comment by leafcutterant** [comment]
> the fiat fee is already shown in the confirmation dialog.
> how many times should it be shown?

@ecdsa It should be shown in both places because this way one has to continuously press "Send" and then go back  just to adjust fees. Other than being time-consuming, this is also frustrating and unnerving because one doesn't like to hit "Send" on something that they don't know the full costs of. This especially applies to new users who don't know how Electrum works. Experienced ones will know that fees will be displayed later and that a password entry will be needed before the transaction is sent, but even then it's unnerving.

**Comment by Engelberg** [comment]
Is the fee ever visible in fiat when saving the transaction for cold signing?  If not, that's another reason for putting the fiat fee earlier.

**Comment by SomberNight** [comment]
For reference, this is how the Send tab looks now by default:
![send_tab_313_2](https://user-images.githubusercontent.com/29142493/39334342-86a75ad6-49ae-11e8-8556-ad29fa033b94.PNG)
And with `Edit fees manually` enabled:
![send_tab_313_1](https://user-images.githubusercontent.com/29142493/39334345-8950fa12-49ae-11e8-9f95-fbaa118f1e07.PNG)

**Comment by SomberNight** [comment]
> I also agree that the transaction size is a technical metric that would confuse many and would do more harm than good in the position you've put it in

The transaction size was added to educate users how a fee is calculated; so that e.g. they are less likely to start asking questions why the absolute fee is so large. _(For instance, there was one user on IRC around November, who was outraged that they were suggested to pay a ~0.12 BTC fee (which is above the 0.1 BTC bitcoin core policy limit so the tx would not even propagate but that's another story). He was trying to send all his coins to an exchange to sell, had many small mining UTXO, and the tx was 10 KB+. He was unaware that the tx size in bytes matters for the fee.)_

**Comment by fresheneesz** [comment]
> The transaction size was added to educate users how a fee is calculated

Help information like that should ideally be easily accessible, but also not front and center. A small mouse-over question mark next to the fee that gives additional information would be better, especially since its currently taking up valuable real estate that could be usd to make the interface easier to use for both new and long-time users.

**Comment by leafcutterant** [comment]
@SomberNight Sorry, I already forgot how the Send tab looks right now. I agree that it's good to display transaction size, the thing I argues was @fresheneesz's placement. I also agree that the `feeRate` × `txSize` = `fee` is visually a good way to explain fees, even if it takes up some space.

Taking all into account, here's my sketch.

* I shrunk the fee rate box to not let the aligned box above it to be irrationally long, but be able to accommodate four digits. (It's fair to expect that fees will never be higher than 9999 sat/b, and beyond 99.99 sat/b, you don't really need decimal precision to be displayed.)
* Likewise, it's fair to expect (or is it?) that there won't be transactions over 99999 bytes (= 97 kB), though this could be expanded to 7 digits (not 8 because we are limited to 4 virtual MBs).
* Expanded the slider to have better precision, not being limited to the shrunk fee rate box.
* I thought about whether it is good to have a fee that is editable on fiat basis or not. It increases the chance for user error, but I ended up up thinking yes because 1) user freedom, 2) a less common way of defining fees exists that goes like "the average tx fee is 25 dollars" and I saw at least one site that gives fee advice in such a way.

![wa](https://user-images.githubusercontent.com/7106231/39401506-a4df0802-4b46-11e8-92da-f1f89213fda7.png)
